nemo/collections/common/data/lhotse/text_adapters.py

```diff
@@ -0,0 +1,83 @@
+import random
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Iterator, Literal
+
+from lhotse.cut.text import TextExample, TextPairExample
+from lhotse.dataset.dataloading import resolve_seed
+from lhotse.utils import Pathlike
+
+from nemo.collections.common.data.lhotse.nemo_adapters import expand_sharded_filepaths
+
+
+@dataclass
+class LhotseTextAdapter:
+    """
+    ``LhotseTextAdapter`` is used to read a text file and wrap
```

Comment:
for the single text file case, can it generalize to instruction following text dataset?

Reply:
I believe so, provided you have question + context + answer together as a single line in the txt file.

Reply:
the difference is the loss_mask should be only on answer but not question and context. do we plan to produce a loss_mask out of the data loader to accommodate that?

Reply:
In that case it would be simpler to add a `JsonTextAdapter` that expects question/answer/context keys in JSON and maps that to `TextExample`. Then it will be straightforward to compute the mask.